front/src/components/ViewMockup.js

```diff
@@ -49,19 +49,66 @@ class ViewMockup extends Component {
         window.location.href = "/mockupsurvey/add/" + this.state.mockup.id
     };
 
+    startButton() {
+        function startRecording(stream) {
+            let recorder = new MediaRecorder(stream)
+            let data = []
+
+            recorder.ondataavailable = event => data.push(event.data)
+            recorder.start()
+
+            let stopped = new Promise((resolve, reject) => {
+                recorder.onstop = resolve
+                recorder.onerror = event => reject(event.name)
+            })
+
+            return stopped.then(() => data)
+        }
+
+        let displayMediaOptions = {
+            video: {
+                cursor: "always"
+            },
+            audio: false
+        }
+        let video = document.getElementById("video")
+        navigator.mediaDevices.getDisplayMedia(displayMediaOptions)
+            .then(stream => {
+                video.srcObject = stream
+                return new Promise(resolve => video.onplaying = resolve)
+            })
+            .then(() => startRecording(video.captureStream()))
+            .then(recordedChunks => {
+                let a = document.createElement('a');
+                a.href = window.URL.createObjectURL(new Blob(recordedChunks, {type: "video/webm"}));
+                a.download = "Recording.webm";
+                a.click();
+            })
+    }
+
+    stopButton() {
+        let video = document.getElementById("video")
+        video.srcObject.getTracks().forEach(track => track.stop())
+    }
+
     render() {
         if (!AuthenticationService.isUserLoggedIn()) {
             return <Redirect to={"/"}/>
         }
         return (
             <div className="bg-light">
+                <video id="video" hidden autoPlay/>
                 <nav className="navbar navbar-expand-lg navbar-light navbar-secondary">
                     <span className="container float-left navbar-breadcrumbs">
-                        <a href="/project">Projects</a> &ensp; / &ensp;
+                        <a href={"/project"}>Projects</a> &ensp; / &ensp;
```

Comment:
those `{ }` brackets are unnecessary here

Reply:
I know, and it's not really relevant here, but I changed it because IntelliJ had a warning here for some reason. But I will change it back.